sale_order_field_advanced/views/sale_order_line_views.xml

```diff
@@ -0,0 +1,81 @@
+<odoo>
+	<record id='view_sale_order_form_inherit_advanced' model='ir.ui.view'>
+		<field name='name'>sale.order.form (in sale_order_field_advanced)</field>
+		<field name='model'>sale.order</field>
+		<field name='inherit_id' ref='sale.view_order_form' />
+		<field name='arch' type='xml'>
+			<xpath expr="//field[@name='order_line']/tree/field[@name='name']" position='after'>
+				<field name='state' invisible='1' />
+				<field name='adv_product_has_process_control' invisible='1' />
+				<field name='adv_reference'
+					optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_is_tailored' optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_characteristics' optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_modification_id' optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_note' optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_complement_id' optional='hide'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+				<field name='adv_manufacturing_state_id' optional='show'
+					attrs="{'readonly': ['|', ('state', 'in', ['done', 'cancel']), ('adv_product_has_process_control', '!=', True)]}" />
+			</xpath>
+		</field>
+	</record>
+
+	<record id='view_sale_order_line_tree_extend' model='ir.ui.view'>
```

Comment:
This is a primary view for `sale.order.line` model, and it could be automatically selected by other views or actions instead of default one. Then, in order to prevent this, please set a low priority for it, a number higher than usual `16` for default views (e.g. `999`)